pkg/px-resource-gateway/px_resource_gateway.go

```diff
@@ -0,0 +1,84 @@
+package server
+
+import (
+	"context"
+
+	pb "github.com/libopenstorage/operator/proto"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/sirupsen/logrus"
+	emptypb "google.golang.org/protobuf/types/known/emptypb"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type server struct {
+	k8s         core.Ops
+	semaphorePQ SemaphorePriorityQueue
+	pb.UnimplementedSemaphoreServiceServer
+}
+
+// TODO implement semaphore map for different resource ids
+func NewServer() *server {
+	s := &server{
+		k8s:         core.Instance(),
+		semaphorePQ: NewSemaphorePriorityQueue(),
+	}
+	return s
+}
+
+func (s *server) AcquireLock(ctx context.Context, req *pb.AcquireLockRequest) (*pb.AcquireLockResponse, error) {
+	// validate request
```

Comment:
check if `req` is nil

Reply:
actually. instead use all the `GetXXX()` accessors from the gRPC proto output. They already manage `nil`. All the values have a `GetXXX()` function. For example use `req.GetResourceId()`, etc.